flocker/node/agents/ebs.py

```diff
@@ -128,15 +106,18 @@ def _run_with_logging(self, *args, **kwargs):
         """
         original = getattr(self, original_name)
         method = getattr(original, method_name)
-        try:
-            return method(*args, **kwargs)
-        except EC2ResponseError as e:
-            BOTO_EC2RESPONSE_ERROR(
-                code=e.code,
-                message=e.message,
-                request_id=e.request_id,
-            ).write()
-            raise
+
+        # Trace IBlockDeviceAPI ``method`` as Eliot Action.
+        with AWS_ACTION(operation=[method_name, args, kwargs]):
+            try:
+                return method(*args, **kwargs)
+            except EC2ResponseError as e:
+                BOTO_EC2RESPONSE_ERROR(
+                    aws_code=e.code,
+                    aws_message=e.message,
+                    aws_request_id=e.request_id,
+                ).write()
```

Comment:
Yeah, here's an example of the resulting duplication:
 * http://build.clusterhq.com/builders/flocker%2Ffunctional%2Faws%2Fcentos-7%2Fstorage-driver/builds/72/steps/trial/logs/test.log
```
2015-05-27 00:24:26+0000 [-] ELIOT: {'aws_code': u'VolumeInUse',
	 'aws_message': u'Volume vol-a346f1b3 is currently attached to i-115ccfe6',
	 'aws_request_id': u'7c9da9e9-ed15-4c3a-878c-4f29a209cfc7',
	 u'message_type': u'boto:boto_ec2response_error',
	 u'task_level': [2],
	 u'task_uuid': u'0a1abfe9-7a66-43d6-949c-6c744fd3f1e0',
	 u'timestamp': 1432686266.464014}
2015-05-27 00:24:26+0000 [-] ELIOT: {u'action_status': u'failed',
	 u'action_type': u'flocker:node:agents:blockdevice:aws',
	 u'exception': u'boto.exception.EC2ResponseError',
	 u'reason': u'EC2ResponseError: 400 Bad Request\n<?xml version="1.0" encoding="UTF-8"?>\n<Response><Errors><Error><Code>VolumeInUse</Code><Message>Volume vol-a346f1b3 is currently attached to i-115ccfe6</Message></Error></Errors><RequestID>7c9da9e9-ed15-4c3a-878c-4f29a209cfc7</RequestID></Response>',
	 u'task_level': [3],
	 u'task_uuid': u'0a1abfe9-7a66-43d6-949c-6c744fd3f1e0',
	 u'timestamp': 1432686266.464498}
```

All the information logged in the message is already logged by the failed action...wrapped in XML tags...but the info is there already.